### DISCUSSION TITLE
Performous 1.2.0

### DISCUSSION THREAD

**Discussion by Baklap4** [performous/performous#702] [Announcements]
Now for real an official new version where we patched a couple of things from the rc1 version.

It's been quite some time we've made a release (almost 6 years). Within this release we've mainly stabilized Performous and focused on automation and solving bugs.
However there are also a few new features within this release!

# Upcoming releases

We've made some big gains in automation. This is why we're going to change the releasing strategy.
With the upcoming release we'll have a release candidate. A release candidate will be a branch on its own (with a tag) where only hotfixes will be applied to.
After a succesful testing period we'll then make a release. If the testing period is not succesful we'll create a new release candidate and repeat the testing.
Once a release candidate is ready we'll make a release branch and tag that branch as the official release.

We'll try to create more releases in less time. Preferably after every PR. As we always did we will use semver as version strategy.

# New features

- Webserver overhaul (@baklap4)
    - _Not enabled on Windows builds._
- Add song paths in game (@nieknooijens)
- Configurable playlist duration (@nieknooijens)
- Audio system overhaul (@Lord-Kamina)
- Added Pro Mode to the drums (@earlye)
- Metadata caching (@baklap4)
- Changing languages from within the menu (@baklap4)
- Windows installer (@baklap4)
- Improved frequency detection while singing (@Tronic)
- Note graph scaling: Static, dynamic (@twollgam)
- Difficulty levels for singing (@isadofschi)
- Proper support for Hi-DPI displays (@Lord-Kamina)
- Auto-merging of duet tracks (@Lord-Kamina)

# Bugfixes

- Remembering sort order (@nieknooijens)
- Fullscreen mode on MacOS (@Tronic)
- Fix a crash in playlist screen when there was an empty playlist (@nieknooijens)
- Several memory leaks (@OznOg)
- Shortcuts in game representation (@baklap4)
- Path screen selection on top (@OznOg)
- Devices were off by one (@OznOg)
- Missing cache entries (@OznOg)
- Autodetecting webcam (@baklap4)
- Pitchwave order is now based on score. Highest gets shown on top (@baklap4)
- Fix a crash where a player gets more than 10k points (@baklap4)
- Nav keys getting stuck (@Tronic)
- Fixed a couple missing glyphs in the `Goblin One` font (@Lord-Kamina)
- Updated version of `Droid Sans` font (@Lord-Kamina)
- Default aspect ratio is now 16:9 (@Lord-Kamina)
- Fix backward audio seeking (@OznOg)
- Better error handling with corrupt files (@OznOg)

# Translations

- Dutch (@baklap4 & @nieknooijens)
- German (@twollgam)
- French (@OznOg)
- Spanish (@Lord-Kamina)
- Hungarian (@skyli91)
- Slovak (@jose1711)
- Japanese (@nieknooijens)

# Other

- using c++ std more (@Tronic & @OznOg)
- Use libxml++3 or libxml++5 when available (@yoda-jm & @baklap4)
- Better unicode handling (@Lord-Kamina)
- Change from OpenGL 2.1 to OpenGL 3.3 and fixed display issues on older intel GPUs (@Tronic, @Lord-Kamina)
- CI + Artifacts (@baklap4, @Lord-Kamina, @deuteragenie, @ooshlablu)
- Native compilation on Windows (@baklap4)
- Lots of cmake fixes and cleanup (@OznOg, @Lord-Kamina, @yoda, @baklap4)
- Refactoring (@twollgam)
- Meson build support (@Tronic, @deuteragenie)
- Using libglm instead of in-house math. (@Lord-Kamina)
- Various fixes in the macOS bundling script (and better support for Homebrew) (@Lord-Kamina, @deuteragenie)

# Notes

On Windows you might want to enable UTF-8 support throughout the system. This way windows can support special characters within song paths. You can enable this by doing this:
1. Windows btn + R -> `intl.cpl`
2. Hit the 2nd tab called "administritive"
3. Hit the button "Change System Locale..."
4. Tick the checkbox "Beta: Use Unicode UTF-8 for worldwide language support"
5. Restart pc

<hr /><em>This discussion was created from the release <a href='https://github.com/performous/performous/releases/tag/1.2.0'>Performous 1.2.0</a>.</em>

**Comment by Tronic** [comment]
New installer needs to be added on https://performous.org/

**Reply by Baklap4** [reply]
Yea, i thought that was added in a pr. But it wasn't. 
I've created https://github.com/performous/web/pull/13 to go to a fully automatic deploy using semaphore
And we've got https://github.com/performous/web/pull/15 to resolve the download section should probably be merged somewhere soon
